[dtaywd] iP

dtaywd opened the pull request.

dtaywd committed:
Add chatbot's name and welcome message

dtaywd committed:
Change name of file

dtaywd committed:
test change in git folder structure

dtaywd committed:
add echo function (level 1)

dtaywd committed:
Add list function

dtaywd committed:
Add task marking function

dtaywd committed:
update task tracking

-able to differentiate tasks (todo, deadline, event)
-implemented task polymorphism (to have the above types of tasks)
-abstracted out user interface
-updated printing logic

dtaywd committed:
Update task printing

error checking for when user gives an unrecognised command and printing out the number of tasks in list

dtaywd committed:
Add automated text ui testing

ACTUAL.TXT matches EXPECTED.TXT however, it does not tell me what the differences are and one still needs to manually inspect the outputs

Ijaaz01 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,36 @@
+    public Parser () {}
+    
+    // no error checking for processing task information has been implemented (e.g. deadline but no /by)
+    public String[] processTaskInformation(String userInput) {
```
I like that you have a separate class to parse information, makes reading code a lot easier!

Ijaaz01 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,33 @@
+public class Task {
+    private int id;
+    private String content;
+    private boolean isDone;
```
Good job naming your boolean variable with the 'is' prefix!

Ijaaz01 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,33 @@
+        this.isDone = isDone;
+    }
+
+    public int getId() {
```
Good job using 'get' prefix for your return function

Ijaaz01 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    @Override
+    public String toString() {
```
Good use of toString override to make your life easier when printing!

genexus85 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+
+    public Deadline(int id, String content, boolean isDone, String by) {
+        super(id, content, isDone);
+        this.by = by;
```
Consider renaming the by variable to something more descriptive, eg dueDate.

genexus85 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,15 @@
+public class Event extends Task{
+    protected String from;
+    protected String to;
```
Similarly, consider naming the variable 'from' and 'to' to something more descriptive, eg. startDate and endDate

genexus85 commented on src/main/java/UserInterface.java:
```diff
@@ -0,0 +1,35 @@
+
+    public void processUserCommand(String userInput) {
+        userInput = userInput.toLowerCase();
+        if (userInput.equals("bye")) {
+            System.out.println("Bye. Hope to see you again soon!");
+        } else if (userInput.equals("list")) {
+            taskManager.showListContents();
+            userInput = myScanner.nextLine();
+            processUserCommand(userInput);
+        } else if (userInput.contains("mark")) {
+            taskManager.changeTaskStatus(userInput);
+            userInput = myScanner.nextLine();
+            processUserCommand(userInput);
+        } else if (userInput.contains("todo") || userInput.contains("event") || userInput.contains("deadline")){
+            System.out.println("Got it. I've added this task:");
+            taskManager.addListContents(userInput);
+            userInput = myScanner.nextLine();
+            processUserCommand(userInput);
+        }
+        else {
+            System.out.println("Please enter a valid command.");
+            userInput = myScanner.nextLine();
+            processUserCommand(userInput);
+        }
```
Consider the use of switch cases instead of if-else blocks to handle each command

genexus85 commented on src/main/java/UserInterface.java:
```diff
@@ -0,0 +1,35 @@
+    }
+
+    public void processUserCommand(String userInput) {
+        userInput = userInput.toLowerCase();
```
I like the idea of normalizing the command into lowercase!

genexus85 reviewed (commented):
Overall, code is well structured. Good job!

dtaywd committed:
Bug fix in testing code

dtaywd committed:
Add error checking to tasks

There are now task specific exceptions in the package exceptions. There are also checks for if the task description/duedate/to/from are empty. A corresponding error message would be printed out.

dtaywd committed:
Add packages and update testing

Tasks package was added ontop of the exceptions package. Updated the tests to include error the newly implemented error checking. Fixed ascii art issue as well. Lastly some of the messages that Sebastian replies with was updated. Changed a magic number to a constant as well.

dtaywd committed:
Merge branch 'branch-Level-5'

dtaywd committed:
Merge branch 'master' of https://github.com/dtaywd/ip

dtaywd committed:
Add error checking for marking/unmarking

dtaywd committed:
Refactored task manager error messages

dtaywd committed:
Replaced the tasklist array with the collection arraylist

dtaywd committed:
Refactored userinterface

Refactored userinterface to not recursively call when getting user inputs and to use a while loop instead

dtaywd committed:
add delete task functionality and some bug fixes

dtaywd committed:
Add file processor

dtaywd committed:
Merge branch 'Level-7'

dtaywd committed:
task can be loaded from data.txt

can add and list tasks and has error checking for no data.txt file. However, the deleting for tasks remains buggy as the file cant be deleted.

sevenseasofbri commented on src/main/java/FileProcessor.java:
```diff
@@ -0,0 +1,97 @@
+import Tasks.Task;
+
+import java.io.*;
```
avoid wildcard imports

sevenseasofbri commented on src/main/java/FileProcessor.java:
```diff
@@ -0,0 +1,97 @@
+
+    public static void printFileContents(String filePath) throws FileNotFoundException {
+        File f = new File(filePath); // create a File for the given file path
+        Scanner s = new Scanner(f); // create a Scanner using the File as the source
```
1. variable names can be more descriptive 
2. The comments are a bit redundant

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#use-name-to-explain

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#do-not-repeat-the-obvious

sevenseasofbri commented on src/main/java/FileProcessor.java:
```diff
@@ -0,0 +1,97 @@
+    public static void removeLineData(String filePath, int deleteLine) {
+        try {
+            File inputFile = new File(filePath);
+            File tempFile = new File("./src/main/temp.txt"); // Temporary file to write modified contents
```
instead of using a magic literal, you can store it in a constant instead 

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers

sevenseasofbri commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,101 @@
+    
+    public Parser () {}
+    
+    // no error checking for processing task information has been implemented (e.g. deadline but no /by)
```
avoid writing comments like they are notes to yourself

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#write-to-the-reader

sevenseasofbri commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,101 @@
+    public Parser () {}
+    
+    // no error checking for processing task information has been implemented (e.g. deadline but no /by)
+    public static String[] processTaskInformation(String userInput) throws TodoLackInputsException, DeadlineLackInputsException, EventLackInputsException {
```
this function is a little longer than what ia accepted (> 30 LOC), see if you can abstract out similar portions of the code into a function 

also layout-wise, i don't think you need to have so much spacing between the cases 

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods

sevenseasofbri commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,101 @@
+        return output;
+    }
+
+    public static int processTaskIdforMarkingAndDeletingTask(String userInput) {
```
similar for this function (refer to my comment on the previous function)

sevenseasofbri commented on src/main/java/Sebastian.java:
```diff
@@ -0,0 +1,15 @@
+public class Sebastian {
+    public static void main(String[] args) {
+        String logo = " ____       _               _   _\n" +
```
maybe this logo can be a class-level constant?

sevenseasofbri commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,222 @@
+            break;
+
+        case ("error"):
+            //should not hit this line
```
maybe avoid writing comments that read as comments to yourself as the programmer

see here: https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#write-to-the-reader

sevenseasofbri commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,222 @@
+    public static boolean isStringInteger(String number ){
+        try{
+            Integer.parseInt(number);
+        }catch(Exception e ){
```
make sure you are consistent with the spacing, in this case it should be:
`} catch(Exception e) {`

sevenseasofbri commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,222 @@
+        } else if (purpose.equals("deleteTask")) {
+            commandStructure = "\"delete x\"";
+        }
+        else {
```
layout of if else blocks should be as follows:
```java
if (condition) {
    statements;
} else if (condition) {
    statements;
} else {
    statements;
}
```
https://se-education.org/guides/conventions/java/basic.html#layout

sevenseasofbri reviewed (commented):
Well written code. Overall spacing and layout to fit coding standard guidelines can be improved. Else, good job! 👍🏽

dtaywd committed:
Add saving functionality for task marking and deleting

dtaywd committed:
Attempted to create jar file

dtaywd committed:
Add and test Jar file

dtaywd committed:
refactor code

changed the name of the filemanger class to storage and the userinterface class to ui

dtaywd committed:
Date and times is now converted to MMM dd yyyy structure

dtaywd committed:
Implement find function

Find gives users a way to find a task by searching for a keyword.

dtaywd committed:
Add Javadoc comments

dtaywd committed:
Merge pull request #1 from dtaywd/heads/Level-8

Date and times is now converted to MMM dd yyyy structure

dtaywd committed:
Merge tag 'Level-9'

dtaywd committed:
Merge branch 'A-JavaDoc'

# Conflicts:
#	src/main/java/Parser.java

dtaywd committed:
Update README.md

dtaywd committed:
remove emojis

dtaywd committed:
update README

dtaywd committed:
add spacing between user inputs

dtaywd committed:
Update README

dtaywd committed:
update README
